zebra-consensus/src/primitives/groth16/params.rs

```diff
@@ -137,7 +136,7 @@ impl Groth16Parameters {
                 sapling_output_path,
             ) {
                 retries += 1;
-                if retries >= PARAMETER_DOWNLOAD_MAX_RETRIES {
+                if retries > PARAMETER_DOWNLOAD_MAX_RETRIES {
```

Comment:
Optional: increment retries after the if statement `retries` is `attempts - 1` when the original condition is checked instead of updating the condition.

Reply:
I'd say the current implementation follows the domain logic well:

- When we enter https://github.com/zcashfoundation/zebra/blob/4208704c92c446422358baa3b89fa561473fff04/zebra-consensus/src/primitives/groth16/params.rs#L186, we already know the initial attempt failed, so
- the initial attempt is regarded as the zeroth retry.
- We then count the retries starting at one, and
- we increment the counter after we've done a retry (not when we're about to do one), so
- we panic once we exceed the max number of retries (not when we're at the limit).